Analytics/Classes/Middlewares/SEGContext.m

```diff
@@ -68,10 +68,10 @@ - (id)copyWithZone:(NSZone *)zone
 {
     SEGContext *ctx = [[SEGContext allocWithZone:zone] initWithAnalytics:self._analytics];
     ctx.eventType = self.eventType;
-    ctx.userId = self.userId;
-    ctx.anonymousId = self.anonymousId;
-    ctx.payload = self.payload;
-    ctx.error = self.error;
+    ctx.userId = [self.userId copy];
+    ctx.anonymousId = [self.anonymousId copy];
+    ctx.payload = self.payload; // FIX: this needs to be copied!!
```

Comment:
What happens if we don't copy it?

Reply:
Nothing, *yet*.  For destination middleware this would be problematic the same payload dictionary would be getting modified all over.